internal/validators/namespace.go

```diff
@@ -107,60 +109,71 @@ func (v *Namespace) handle(req *nsRequest) admission.Response {
 }
 
 // illegalIncludedNamespaceLabel checks if there's any illegal use of the
-// included-namespace label on namespaces.
-// TODO these validating rules will be removed and converted into a mutating
-//  webhook. See https://github.com/kubernetes-sigs/hierarchical-namespaces/issues/37
+// included-namespace label on namespaces. It only checks a Create or an Update
+// request, so the newns is not empty in the request.
 func (v *Namespace) illegalIncludedNamespaceLabel(req *nsRequest) admission.Response {
-	labelValue, hasLabel := req.ns.Labels[api.LabelIncludedNamespace]
-	isExcluded := config.ExcludedNamespaces[req.ns.Name]
-
-	// An excluded namespace should not have the included-namespace label.
-	//
-	// Note: this only blocks the request if the excluded namespace is newly
-	// created or is updated with a newly added included-namespace label because
-	// existing illegal included-namespace label should have already been removed
-	// by our reconciler. For example, even when the VWHConfiguration is removed,
-	// adding the label to an excluded namespace would pass but the label is
-	// immediately removed; when the VWHConfiguration is there but the reconcilers
-	// are down, any request gets denied anyway.
+	labelValue, hasLabel := req.newns.Labels[api.LabelIncludedNamespace]
+	var oldLabelValue string
+	var oldHasLabel bool
+	if req.oldns != nil {
+		oldLabelValue, oldHasLabel = req.oldns.Labels[api.LabelIncludedNamespace]
```

Comment:
I would have probably just put this here:

```
if req.oldns != nil && req.oldns.Labels[api.LabelIncludedNamespace] == labelValue {
  return allow("")
}
```

Then, after this line, you know that something's changed so you could leave the rest of the logic alone. Up to you if you want to revert to that.

Reply:
Hmm, why we allow when `req.oldns.Labels[api.LabelIncludedNamespace] == labelValue`? The requested (new) namespace instance can change to have no/wrong label for included namespace or have label on excluded namespace, which should both be rejected.

Reply:
Discussed offline, I mistook the `labelValue` as the correct value.
Adrian's suggestion sgtm, will make that update.